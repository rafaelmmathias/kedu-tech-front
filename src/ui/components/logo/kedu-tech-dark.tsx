import { theme } from "kedu-tech-ui";
import { SVGLogo } from "./svg-logo";

interface KeduTechDarkProps {
  width: number;
}

export const KeduTechDark: React.FC<KeduTechDarkProps> = ({ width }) => {
  const { useToken, defaultAlgorithm } = theme;
  const { token } = useToken();
  const lightTheme = defaultAlgorithm(token);

  const fillPrimaryColor = lightTheme.colorPrimaryBg;
  const fillSecondaryColor = lightTheme.colorPrimary;

  return (
    <SVGLogo
      width={width}
      primaryColor={fillPrimaryColor}
      secondaryColor={fillSecondaryColor}
    />
  );
};
