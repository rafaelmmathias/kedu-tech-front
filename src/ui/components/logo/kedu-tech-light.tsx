import { theme } from "kedu-tech-ui";
import { SVGLogo } from "./svg-logo";

interface KeduTechLightProps {
  width: number;
}

export const KeduTechLight: React.FC<KeduTechLightProps> = ({ width }) => {
  const { useToken, darkAlgorithm } = theme;
  const { token } = useToken();
  const darkTheme = darkAlgorithm(token);

  const fillPrimaryColor = darkTheme.colorPrimaryBg;
  const fillSecondaryColor = darkTheme.colorPrimary;

  return (
    <SVGLogo
      width={width}
      primaryColor={fillPrimaryColor}
      secondaryColor={fillSecondaryColor}
    />
  );
};
