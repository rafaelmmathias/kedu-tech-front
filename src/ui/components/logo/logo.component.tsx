import { Row, theme as themeUI } from "kedu-tech-ui";
import { KeduTechDark } from "./kedu-tech-dark";
import { KeduTechLight } from "./kedu-tech-light";

const LIGHT_THEME_INDEX = 0;
const DARK_THEME_INDEX = 1;
const DEFAULT_WIDTH = 175;

interface LogoProps {
  width?: number;
  theme?: "light" | "dark";
}

export const Logo: React.FC<LogoProps> = ({ width = DEFAULT_WIDTH, theme }) => {
  const { useToken } = themeUI;
  const themes = [KeduTechLight, KeduTechDark];
  const themesMap = { light: LIGHT_THEME_INDEX, dark: DARK_THEME_INDEX };

  const {
    theme: { id },
  } = useToken();

  let ThemedLogo = themes[id] || KeduTechDark;

  if (theme) {
    ThemedLogo = themes[themesMap[theme]];
  }

  return (
    <Row justify={"center"}>
      <ThemedLogo width={width} />
    </Row>
  );
};
