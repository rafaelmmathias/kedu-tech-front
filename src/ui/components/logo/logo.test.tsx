import renderer from "react-test-renderer";
import { Logo } from "./logo.component";

describe("Logo", () => {
  it("should match snapshot with default behavior", () => {
    const tree = renderer.create(<Logo />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("should match snapshot with dark theme", () => {
    const tree = renderer.create(<Logo theme="dark" />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
