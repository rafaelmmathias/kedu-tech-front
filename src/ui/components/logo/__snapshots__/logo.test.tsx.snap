// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Logo should match snapshot with dark theme 1`] = `
<div
  className="ant-row ant-row-center css-dev-only-do-not-override-1me4733"
  style={Object {}}
>
  <svg
    fill="none"
    viewBox="0 0 529 226"
    width={175}
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      d="M528.5 96.5L487 150H463L506.5 96.5L463 43H487L528.5 96.5Z"
      fill="#1677ff"
    />
    <path
      d="M27 96.5L68.5001 43H92.5L49 96.5L92.5 150H68.5001L27 96.5Z"
      fill="#1677ff"
    />
    <path
      d="M0 0H21V150H0V0Z"
      fill="#1677ff"
    />
    <path
      d="M331.7 167.4L322.7 170.1V180H312.8V188.55H322.7V225H331.7V188.55H342.5V180H331.7V167.4Z"
      fill="#e6f4ff"
    />
    <path
      d="M351.565 198.45C352.405 195.27 354.085 192.72 356.605 190.8C359.185 188.82 362.215 187.83 365.695 187.83C369.175 187.83 372.205 188.82 374.785 190.8C377.305 192.72 378.985 195.27 379.825 198.45H351.565ZM389.005 206.55C389.245 204.93 389.365 203.58 389.365 202.5C389.365 195.72 387.115 190.14 382.615 185.76C378.175 181.32 372.535 179.1 365.695 179.1C358.855 179.1 353.185 181.32 348.685 185.76C344.245 190.14 342.025 195.72 342.025 202.5C342.025 209.28 344.245 214.86 348.685 219.24C353.185 223.68 358.855 225.9 365.695 225.9C370.915 225.9 375.505 224.55 379.465 221.85C383.365 219.15 386.155 215.55 387.835 211.05H377.845C375.025 215.13 370.975 217.17 365.695 217.17C362.215 217.17 359.185 216.18 356.605 214.2C354.085 212.28 352.405 209.73 351.565 206.55H389.005Z"
      fill="#e6f4ff"
    />
    <path
      d="M430.115 206.1C429.395 209.4 427.745 212.07 425.165 214.11C422.585 216.15 419.495 217.17 415.895 217.17C411.635 217.17 408.125 215.79 405.365 213.03C402.605 210.27 401.225 206.76 401.225 202.5C401.225 198.24 402.605 194.73 405.365 191.97C408.125 189.21 411.635 187.83 415.895 187.83C419.495 187.83 422.585 188.85 425.165 190.89C427.745 192.93 429.395 195.6 430.115 198.9H439.205C438.425 193.08 435.905 188.34 431.645 184.68C427.325 180.96 422.075 179.1 415.895 179.1C409.055 179.1 403.415 181.32 398.975 185.76C394.475 190.14 392.225 195.72 392.225 202.5C392.225 209.28 394.475 214.86 398.975 219.24C403.415 223.68 409.055 225.9 415.895 225.9C422.075 225.9 427.325 224.04 431.645 220.32C435.905 216.66 438.425 211.92 439.205 206.1H430.115Z"
      fill="#e6f4ff"
    />
    <path
      d="M468.623 179.1C462.383 179.1 457.433 181.32 453.773 185.76V162H444.773V225H453.773V200.25C453.773 196.35 454.913 193.29 457.193 191.07C459.413 188.79 462.473 187.65 466.373 187.65C470.273 187.65 473.333 188.79 475.553 191.07C477.833 193.29 478.973 196.35 478.973 200.25V225H487.973V199.35C487.973 193.47 486.203 188.64 482.663 184.86C479.063 181.02 474.383 179.1 468.623 179.1Z"
      fill="#e6f4ff"
    />
    <path
      d="M118.705 86.7042C120.704 79.1399 124.703 73.0742 130.701 68.507C136.841 63.7972 144.052 61.4423 152.335 61.4423C160.617 61.4423 167.829 63.7972 173.969 68.507C179.967 73.0742 183.965 79.1399 185.964 86.7042H118.705ZM207.813 105.972C208.384 102.118 208.67 98.907 208.67 96.338C208.67 80.2103 203.315 66.9371 192.604 56.5183C182.037 45.9568 168.614 40.6761 152.335 40.6761C136.056 40.6761 122.561 45.9568 111.851 56.5183C101.284 66.9371 96 80.2103 96 96.338C96 112.466 101.284 125.739 111.851 136.158C122.561 146.719 136.056 152 152.335 152C164.758 152 175.683 148.789 185.107 142.366C194.389 135.944 201.03 127.38 205.028 116.676H181.252C174.54 126.381 164.901 131.234 152.335 131.234C144.052 131.234 136.841 128.879 130.701 124.169C124.703 119.602 120.704 113.536 118.705 105.972H207.813Z"
      fill="#e6f4ff"
    />
    <path
      d="M272.239 131.662C261.814 131.662 253.318 128.379 246.749 121.814C240.18 115.249 236.895 106.757 236.895 96.338C236.895 85.9193 240.18 77.4272 246.749 70.862C253.318 64.2967 261.814 61.0141 272.239 61.0141C282.663 61.0141 291.16 64.2967 297.728 70.862C304.297 77.4272 307.582 85.9193 307.582 96.338C307.582 106.757 304.297 115.249 297.728 121.814C291.16 128.379 282.663 131.662 272.239 131.662ZM307.582 0V56.9465C297.3 46.0995 284.448 40.6761 269.026 40.6761C253.746 40.6761 241.037 45.9568 230.898 56.5183C220.616 67.2225 215.475 80.4958 215.475 96.338C215.475 112.18 220.616 125.454 230.898 136.158C241.037 146.719 253.746 152 269.026 152C284.448 152 297.3 146.577 307.582 135.73V149.859H329.002V0H307.582Z"
      fill="#e6f4ff"
    />
    <path
      d="M428.58 101.69C428.58 110.967 425.867 118.246 420.44 123.527C415.157 128.95 407.874 131.662 398.592 131.662C389.31 131.662 382.027 128.95 376.743 123.527C371.317 118.246 368.604 110.967 368.604 101.69V42.8169H347.184V103.831C347.184 117.818 351.396 129.307 359.822 138.299C368.39 147.433 379.528 152 393.237 152C408.088 152 419.869 146.648 428.58 135.944V149.859H450V42.8169H428.58V101.69Z"
      fill="#e6f4ff"
    />
  </svg>
</div>
`;

exports[`Logo should match snapshot with default behavior 1`] = `
<div
  className="ant-row ant-row-center css-dev-only-do-not-override-1me4733"
  style={Object {}}
>
  <svg
    fill="none"
    viewBox="0 0 529 226"
    width={175}
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      d="M528.5 96.5L487 150H463L506.5 96.5L463 43H487L528.5 96.5Z"
      fill="#1668dc"
    />
    <path
      d="M27 96.5L68.5001 43H92.5L49 96.5L92.5 150H68.5001L27 96.5Z"
      fill="#1668dc"
    />
    <path
      d="M0 0H21V150H0V0Z"
      fill="#1668dc"
    />
    <path
      d="M331.7 167.4L322.7 170.1V180H312.8V188.55H322.7V225H331.7V188.55H342.5V180H331.7V167.4Z"
      fill="#111a2c"
    />
    <path
      d="M351.565 198.45C352.405 195.27 354.085 192.72 356.605 190.8C359.185 188.82 362.215 187.83 365.695 187.83C369.175 187.83 372.205 188.82 374.785 190.8C377.305 192.72 378.985 195.27 379.825 198.45H351.565ZM389.005 206.55C389.245 204.93 389.365 203.58 389.365 202.5C389.365 195.72 387.115 190.14 382.615 185.76C378.175 181.32 372.535 179.1 365.695 179.1C358.855 179.1 353.185 181.32 348.685 185.76C344.245 190.14 342.025 195.72 342.025 202.5C342.025 209.28 344.245 214.86 348.685 219.24C353.185 223.68 358.855 225.9 365.695 225.9C370.915 225.9 375.505 224.55 379.465 221.85C383.365 219.15 386.155 215.55 387.835 211.05H377.845C375.025 215.13 370.975 217.17 365.695 217.17C362.215 217.17 359.185 216.18 356.605 214.2C354.085 212.28 352.405 209.73 351.565 206.55H389.005Z"
      fill="#111a2c"
    />
    <path
      d="M430.115 206.1C429.395 209.4 427.745 212.07 425.165 214.11C422.585 216.15 419.495 217.17 415.895 217.17C411.635 217.17 408.125 215.79 405.365 213.03C402.605 210.27 401.225 206.76 401.225 202.5C401.225 198.24 402.605 194.73 405.365 191.97C408.125 189.21 411.635 187.83 415.895 187.83C419.495 187.83 422.585 188.85 425.165 190.89C427.745 192.93 429.395 195.6 430.115 198.9H439.205C438.425 193.08 435.905 188.34 431.645 184.68C427.325 180.96 422.075 179.1 415.895 179.1C409.055 179.1 403.415 181.32 398.975 185.76C394.475 190.14 392.225 195.72 392.225 202.5C392.225 209.28 394.475 214.86 398.975 219.24C403.415 223.68 409.055 225.9 415.895 225.9C422.075 225.9 427.325 224.04 431.645 220.32C435.905 216.66 438.425 211.92 439.205 206.1H430.115Z"
      fill="#111a2c"
    />
    <path
      d="M468.623 179.1C462.383 179.1 457.433 181.32 453.773 185.76V162H444.773V225H453.773V200.25C453.773 196.35 454.913 193.29 457.193 191.07C459.413 188.79 462.473 187.65 466.373 187.65C470.273 187.65 473.333 188.79 475.553 191.07C477.833 193.29 478.973 196.35 478.973 200.25V225H487.973V199.35C487.973 193.47 486.203 188.64 482.663 184.86C479.063 181.02 474.383 179.1 468.623 179.1Z"
      fill="#111a2c"
    />
    <path
      d="M118.705 86.7042C120.704 79.1399 124.703 73.0742 130.701 68.507C136.841 63.7972 144.052 61.4423 152.335 61.4423C160.617 61.4423 167.829 63.7972 173.969 68.507C179.967 73.0742 183.965 79.1399 185.964 86.7042H118.705ZM207.813 105.972C208.384 102.118 208.67 98.907 208.67 96.338C208.67 80.2103 203.315 66.9371 192.604 56.5183C182.037 45.9568 168.614 40.6761 152.335 40.6761C136.056 40.6761 122.561 45.9568 111.851 56.5183C101.284 66.9371 96 80.2103 96 96.338C96 112.466 101.284 125.739 111.851 136.158C122.561 146.719 136.056 152 152.335 152C164.758 152 175.683 148.789 185.107 142.366C194.389 135.944 201.03 127.38 205.028 116.676H181.252C174.54 126.381 164.901 131.234 152.335 131.234C144.052 131.234 136.841 128.879 130.701 124.169C124.703 119.602 120.704 113.536 118.705 105.972H207.813Z"
      fill="#111a2c"
    />
    <path
      d="M272.239 131.662C261.814 131.662 253.318 128.379 246.749 121.814C240.18 115.249 236.895 106.757 236.895 96.338C236.895 85.9193 240.18 77.4272 246.749 70.862C253.318 64.2967 261.814 61.0141 272.239 61.0141C282.663 61.0141 291.16 64.2967 297.728 70.862C304.297 77.4272 307.582 85.9193 307.582 96.338C307.582 106.757 304.297 115.249 297.728 121.814C291.16 128.379 282.663 131.662 272.239 131.662ZM307.582 0V56.9465C297.3 46.0995 284.448 40.6761 269.026 40.6761C253.746 40.6761 241.037 45.9568 230.898 56.5183C220.616 67.2225 215.475 80.4958 215.475 96.338C215.475 112.18 220.616 125.454 230.898 136.158C241.037 146.719 253.746 152 269.026 152C284.448 152 297.3 146.577 307.582 135.73V149.859H329.002V0H307.582Z"
      fill="#111a2c"
    />
    <path
      d="M428.58 101.69C428.58 110.967 425.867 118.246 420.44 123.527C415.157 128.95 407.874 131.662 398.592 131.662C389.31 131.662 382.027 128.95 376.743 123.527C371.317 118.246 368.604 110.967 368.604 101.69V42.8169H347.184V103.831C347.184 117.818 351.396 129.307 359.822 138.299C368.39 147.433 379.528 152 393.237 152C408.088 152 419.869 146.648 428.58 135.944V149.859H450V42.8169H428.58V101.69Z"
      fill="#111a2c"
    />
  </svg>
</div>
`;
